src/libs/TransactionUtils.ts

```diff
@@ -579,11 +579,10 @@ function hasRoute(transaction: OnyxEntry<Transaction>, isDistanceRequestType?: b
 }
 
 function getAllReportTransactions(reportID?: string, transactions?: OnyxCollection<Transaction>): Transaction[] {
-    // `reportID` from the `/CreateDistanceRequest` endpoint return's number instead of string for created `transaction`.
-    // For reference, https://github.com/Expensify/App/pull/26536#issuecomment-1703573277.
-    // We will update this in a follow-up Issue. According to this comment: https://github.com/Expensify/App/pull/26536#issuecomment-1703591019.
-    const nonNullableTransactions: Transaction[] = Object.values(transactions ?? allTransactions ?? {}).filter((transaction): transaction is Transaction => !!transaction);
-    return nonNullableTransactions.filter((transaction) => `${transaction.reportID}` === `${reportID}`);
+    const reportTransactions: Transaction[] = Object.values(transactions ?? allTransactions ?? {}).filter(
+        (transaction): transaction is Transaction => !!transaction && transaction.reportID === reportID,
```

Comment:
I know they both should always be strings `transaction.reportID === reportID` but changin it to this makes a bit worried, should we loosen the equal sign to `==`?

Reply:
Fair point, however this would give us only the first gain from removing unnecessary filter. I am fine with merging it with `==` now, and maybe let's do some more tests with `===` as a separate change so we can get more confidence around it?

Reply:
Hmm okay, lets do `===` it should be all string